File: objects/torrent.go
package objects

import (
	"time"

	"github.com/jinzhu/gorm"
)

type Torrent struct {
	gorm.Model
	CurrentTorrentID uint
	TorrentId        string
	Name             string
	Link             string
	DownloadDir      string
	Seeders          int
	PercentDone      float64
	TotalSize        int64
	ETA              time.Time
	RateDownload     int64
	RateUpload       int64
	Status           int
}

File: server/ui/src/types/torrent.ts
type Torrent = {
    ID: number,
    CreatedAt: Date,
    FailedTorrentID: number,
    TorrentId: string,
    Failed :boolean,
    Name: string,
    Link: string,
    DownloadDir: string,
    Seeders: number,
    PercentDone :number,
    TotalSize :number,
    RateUpload :number,
    RateDownload :number,
    ETA :Date,
};

export default Torrent;
